test/e2e/common/pods.go

```diff
@@ -241,36 +229,16 @@ var _ = framework.KubeDescribe("Pods", func() {
 				if kubeletPod.ObjectMeta.DeletionTimestamp == nil {
 					framework.Logf("deletion has not yet been observed")
 					return false, nil
+				} else if kubeletPod.Name == pod.Name {
+					lastPod = &kubeletPod
```

Comment:
there's no guarantee the pod still exists when we do this list. lastPod could easily be left nil, causing a panic on line 242, etc

Reply:
Yes, that's why added check  `kubeletPod.Name == pod.Name`. It will avoid causing panic.

Reply:
this check will never be false, we already verified this on line 226:

```
if pod.Name != kubeletPod.Name {
  continue
}
```

if the pod was deleted before we saw it a single time in this polling loop, a panic would occurs here:
```
Expect(lastPod.DeletionTimestamp).ToNot(BeNil())
```

Reply:
`else if kubeletPod.Name == pod.Name`
 from this code, if we didn't get the pod which is created by us, lastPod will be nil

if pod found it should not be nil.

Reply:
> if we didn't get the pod which is created by us, lastPod will be nil

correct, which will cause `Expect(lastPod.DeletionTimestamp).ToNot(BeNil())` to panic